lib/base-compiler.js

```diff
@@ -636,7 +637,6 @@ Please select another pass or change filters.`;
     }
 
     initialise() {
-        if (!this.asm) this.asm = new asm.AsmParser(this.compilerProps);
```

Comment:
I'm trying to remember why I did it this way. I think back in the "bad old days" there was some lazy set up of compilerProps (maybe?). Anyway, I'm glad you cleaned this up, I'll stop looking for excuses for my bad coding practices!